File: src/stores/format.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'

export interface FormatRequirement {
  // 页面设置
  paperSize: string
  marginTop: number
  marginBottom: number
  marginLeft: number
  marginRight: number
  gutter: number
  gutterPosition: 'left' | 'top'
  orientation: 'portrait' | 'landscape'
  
  // 字体格式
  bodyFont: string
  bodyFontSize: string
  bodyLineHeight: string
  bodyParagraphSpacing: { before: number; after: number }
  
  heading1: { font: string; fontSize: string; spacing: { before: number; after: number } }
  heading2: { font: string; fontSize: string; spacing: { before: number; after: number } }
  heading3: { font: string; fontSize: string; spacing: { before: number; after: number } }
  heading4: { font: string; fontSize: string; spacing: { before: number; after: number } }
  
  // 页眉页脚
  header: { content: string; font: string; fontSize: string }
  footer: { position: string; format: string }
  
  // 表格格式
  tableFont: string
  tableFontSize: string
  tableBorder: string
  tableAlign: string
}

export interface FormatDiff {
  category: string
  name: string
  current: string
  target: string
}

export interface FormatTemplate {
  id: string
  name: string
  description: string
  format: FormatRequirement
  isPreset: boolean
  createdAt: string
  lastUsedAt?: string
  tags?: string[]
}

export const useFormatStore = defineStore('format', () => {
  // 当前格式要求
  const currentRequirement = ref<FormatRequirement | null>(null)
  
  // 格式差异列表
  const formatDiffs = ref<FormatDiff[]>([])
  
  // 模板列表
  const templates = ref<FormatTemplate[]>([])
  
  // 当前文档路径
  const currentDocument = ref<string>('')
  
  // 招标文件路径
  const tenderDocument = ref<string>('')
  
  // 来源类型
  const sourceType = ref<'tender' | 'template'>('tender')
  
  // 选中的模板
  const selectedTemplate = ref<string>('')
  
  // 输出方式
  const outputMode = ref<'modify' | 'new' | 'copy'>('new')
  
  // 设置格式要求
  function setFormatRequirement(format: FormatRequirement) {
    currentRequirement.value = format
  }
  
  // 设置格式差异
  function setFormatDiffs(diffs: FormatDiff[]) {
    formatDiffs.value = diffs
  }
  
  // 设置当前文档
  function setCurrentDocument(path: string) {
    currentDocument.value = path
  }
  
  // 设置招标文件
  function setTenderDocument(path: string) {
    tenderDocument.value = path
  }
  
  // 添加模板
  function addTemplate(template: FormatTemplate) {
    templates.value.push(template)
  }
  
  // 删除模板
  function removeTemplate(id: string) {
    templates.value = templates.value.filter(t => t.id !== id)
  }
  
  // 更新模板
  function updateTemplate(id: string, template: Partial<FormatTemplate>) {
    const index = templates.value.findIndex(t => t.id === id)
    if (index !== -1) {
      templates.value[index] = { ...templates.value[index], ...template }
    }
  }
  
  // 重置状态
  function reset() {
    currentRequirement.value = null
    formatDiffs.value = []
    currentDocument.value = ''
    tenderDocument.value = ''
    sourceType.value = 'tender'
    selectedTemplate.value = ''
  }
  
  return {
    currentRequirement,
    formatDiffs,
    templates,
    currentDocument,
    tenderDocument,
    sourceType,
    selectedTemplate,
    outputMode,
    setFormatRequirement,
    setFormatDiffs,
    setCurrentDocument,
    setTenderDocument,
    addTemplate,
    removeTemplate,
    updateTemplate,
    reset
  }
})

File: src-tauri/src/models/format.rs
use serde::{Deserialize, Serialize};

/// 格式要求
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FormatRequirement {
    // 页面设置
    pub paper_size: String,
    pub margin_top: f64,
    pub margin_bottom: f64,
    pub margin_left: f64,
    pub margin_right: f64,
    pub gutter: f64,
    pub gutter_position: String,
    pub orientation: String,
    
    // 字体格式
    pub body_font: String,
    pub body_font_size: String,
    pub body_line_height: String,
    pub body_paragraph_spacing: ParagraphSpacing,
    
    pub heading1: HeadingFormat,
    pub heading2: HeadingFormat,
    pub heading3: HeadingFormat,
    pub heading4: HeadingFormat,
    
    // 页眉页脚
    pub header: HeaderFormat,
    pub footer: FooterFormat,
    
    // 表格格式
    pub table_font: String,
    pub table_font_size: String,
    pub table_border: String,
    pub table_align: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ParagraphSpacing {
    pub before: f64,
    pub after: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HeadingFormat {
    pub font: String,
    pub font_size: String,
    pub spacing: ParagraphSpacing,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HeaderFormat {
    pub content: String,
    pub font: String,
    pub font_size: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FooterFormat {
    pub position: String,
    pub format: String,
}

impl Default for FormatRequirement {
    fn default() -> Self {
        Self {
            paper_size: "A4".to_string(),
            margin_top: 2.54,
            margin_bottom: 2.54,
            margin_left: 3.17,
            margin_right: 3.17,
            gutter: 0.0,
            gutter_position: "left".to_string(),
            orientation: "portrait".to_string(),
            
            body_font: "宋体".to_string(),
            body_font_size: "小四".to_string(),
            body_line_height: "1.5倍".to_string(),
            body_paragraph_spacing: ParagraphSpacing { before: 0.0, after: 0.0 },
            
            heading1: HeadingFormat {
                font: "黑体".to_string(),
                font_size: "三号".to_string(),
                spacing: ParagraphSpacing { before: 0.5, after: 0.5 },
            },
            heading2: HeadingFormat {
                font: "黑体".to_string(),
                font_size: "四号".to_string(),
                spacing: ParagraphSpacing { before: 0.5, after: 0.5 },
            },
            heading3: HeadingFormat {
                font: "黑体".to_string(),
                font_size: "小四".to_string(),
                spacing: ParagraphSpacing { before: 0.0, after: 0.0 },
            },
            heading4: HeadingFormat {
                font: "黑体".to_string(),
                font_size: "小四".to_string(),
                spacing: ParagraphSpacing { before: 0.0, after: 0.0 },
            },
            
            header: HeaderFormat {
                content: "项目名称".to_string(),
                font: "宋体".to_string(),
                font_size: "小五".to_string(),
            },
            footer: FooterFormat {
                position: "居中".to_string(),
                format: "第X页 共X页".to_string(),
            },
            
            table_font: "宋体".to_string(),
            table_font_size: "小四".to_string(),
            table_border: "单线".to_string(),
            table_align: "居中".to_string(),
        }
    }
}
